### ISSUE TITLE
Could not load kubeconfig file /var/lib/kubelet/kubeconfig when using vagrant to provision single node

### ISSUE THREAD

**Issue by Leorowe** [coreos/coreos-kubernetes#393]
Hi,

I try to provision single node according this [guild](https://coreos.com/kubernetes/docs/latest/kubernetes-on-vagrant-single.html#start-the-machine) . When executing the `kubectl get nodes`, I got that error:`The connection to the server 172.17.4.99:443 was refused - did you specify the right host or port?`.

using MacOS 10.11.4,  Virtualbox 5.0.16r105874 and vagrant 1.7.4

I use the `journalctl -u kubelet` command to inspect and got the following logs. It seems that **kubeconfig** can't be found during starting the kubelet. 
```
Apr 10 07:50:18 localhost kubelet-wrapper[1310]: Quay.io ACI Converter (ACI conversion signing key) <support@quay.io>
Apr 10 07:50:18 localhost kubelet-wrapper[1310]: Trusting "https://quay.io/aci-signing-key" for prefix "quay.io/coreos/hyperkube" without fingerprint review.
Apr 10 07:50:18 localhost kubelet-wrapper[1310]: Added key for prefix "quay.io/coreos/hyperkube" at "/etc/rkt/trustedkeys/prefix.d/quay.io/coreos/hyperkube/bff313cdaa560b16a8987b8f72abf5f6799d33bc"
Apr 10 07:50:18 localhost kubelet-wrapper[1310]: image: downloading signature from https://quay.io/c1/aci/quay.io/coreos/hyperkube/v1.2.0_coreos.1/aci.asc/linux/amd64/
Apr 10 07:50:18 localhost kubelet-wrapper[1310]: Downloading signature:  0 B/473 B
Apr 10 07:50:18 localhost kubelet-wrapper[1310]: Downloading signature:  473 B/473 B
Apr 10 07:50:20 localhost kubelet-wrapper[1310]: Downloading ACI:  0 B/87.7 MB
Apr 10 07:50:20 localhost kubelet-wrapper[1310]: Downloading ACI:  16.4 KB/87.7 MB
Apr 10 07:50:21 localhost kubelet-wrapper[1310]: Downloading ACI:  277 KB/87.7 MB
Apr 10 07:50:22 localhost kubelet-wrapper[1310]: Downloading ACI:  1.06 MB/87.7 MB
Apr 10 07:50:23 localhost kubelet-wrapper[1310]: Downloading ACI:  2.12 MB/87.7 MB
Apr 10 07:50:25 localhost kubelet-wrapper[1310]: Downloading ACI:  3.36 MB/87.7 MB
Apr 10 07:50:26 localhost kubelet-wrapper[1310]: Downloading ACI:  4.61 MB/87.7 MB
Apr 10 07:50:27 localhost kubelet-wrapper[1310]: Downloading ACI:  5.85 MB/87.7 MB
Apr 10 07:50:28 localhost kubelet-wrapper[1310]: Downloading ACI:  7.12 MB/87.7 MB
Apr 10 07:50:29 localhost kubelet-wrapper[1310]: Downloading ACI:  8.39 MB/87.7 MB
Apr 10 07:50:30 localhost kubelet-wrapper[1310]: Downloading ACI:  9.68 MB/87.7 MB
Apr 10 07:50:31 localhost kubelet-wrapper[1310]: Downloading ACI:  11 MB/87.7 MB
Apr 10 07:50:32 localhost kubelet-wrapper[1310]: Downloading ACI:  12.3 MB/87.7 MB
Apr 10 07:50:33 localhost kubelet-wrapper[1310]: Downloading ACI:  13.6 MB/87.7 MB
Apr 10 07:50:34 localhost kubelet-wrapper[1310]: Downloading ACI:  14.9 MB/87.7 MB
Apr 10 07:50:35 localhost kubelet-wrapper[1310]: Downloading ACI:  16.2 MB/87.7 MB
Apr 10 07:50:36 localhost kubelet-wrapper[1310]: Downloading ACI:  17.7 MB/87.7 MB
Apr 10 07:50:38 localhost kubelet-wrapper[1310]: Downloading ACI:  19.2 MB/87.7 MB
Apr 10 07:50:39 localhost kubelet-wrapper[1310]: Downloading ACI:  20.9 MB/87.7 MB
Apr 10 07:50:40 localhost kubelet-wrapper[1310]: Downloading ACI:  22.7 MB/87.7 MB
Apr 10 07:50:41 localhost kubelet-wrapper[1310]: Downloading ACI:  24.4 MB/87.7 MB
Apr 10 07:50:42 localhost kubelet-wrapper[1310]: Downloading ACI:  25.9 MB/87.7 MB
Apr 10 07:50:43 localhost kubelet-wrapper[1310]: Downloading ACI:  27.5 MB/87.7 MB
Apr 10 07:50:44 localhost kubelet-wrapper[1310]: Downloading ACI:  29.1 MB/87.7 MB
Apr 10 07:50:45 localhost kubelet-wrapper[1310]: Downloading ACI:  30.7 MB/87.7 MB
Apr 10 07:50:46 localhost kubelet-wrapper[1310]: Downloading ACI:  32.3 MB/87.7 MB
Apr 10 07:50:47 localhost kubelet-wrapper[1310]: Downloading ACI:  33.9 MB/87.7 MB
Apr 10 07:50:48 localhost kubelet-wrapper[1310]: Downloading ACI:  35.5 MB/87.7 MB
Apr 10 07:50:49 localhost kubelet-wrapper[1310]: Downloading ACI:  37.1 MB/87.7 MB
Apr 10 07:50:50 localhost kubelet-wrapper[1310]: Downloading ACI:  38.7 MB/87.7 MB
Apr 10 07:50:52 localhost kubelet-wrapper[1310]: Downloading ACI:  40.4 MB/87.7 MB
Apr 10 07:50:53 localhost kubelet-wrapper[1310]: Downloading ACI:  42.1 MB/87.7 MB
Apr 10 07:50:54 localhost kubelet-wrapper[1310]: Downloading ACI:  44.1 MB/87.7 MB
Apr 10 07:50:55 localhost kubelet-wrapper[1310]: Downloading ACI:  46.7 MB/87.7 MB
Apr 10 07:50:56 localhost kubelet-wrapper[1310]: Downloading ACI:  49.9 MB/87.7 MB
Apr 10 07:50:57 localhost kubelet-wrapper[1310]: Downloading ACI:  53.6 MB/87.7 MB
Apr 10 07:50:58 localhost kubelet-wrapper[1310]: Downloading ACI:  58 MB/87.7 MB
Apr 10 07:50:59 localhost kubelet-wrapper[1310]: Downloading ACI:  63 MB/87.7 MB
Apr 10 07:51:00 localhost kubelet-wrapper[1310]: Downloading ACI:  68.2 MB/87.7 MB
Apr 10 07:51:01 localhost kubelet-wrapper[1310]: Downloading ACI:  73.4 MB/87.7 MB
Apr 10 07:51:02 localhost kubelet-wrapper[1310]: Downloading ACI:  78.6 MB/87.7 MB
Apr 10 07:51:03 localhost kubelet-wrapper[1310]: Downloading ACI:  83.9 MB/87.7 MB
Apr 10 07:51:04 localhost kubelet-wrapper[1310]: Downloading ACI:  87.7 MB/87.7 MB
Apr 10 07:51:05 localhost kubelet-wrapper[1310]: image: signature verified:
Apr 10 07:51:05 localhost kubelet-wrapper[1310]: Quay.io ACI Converter (ACI conversion signing key) <support@quay.io>
Apr 10 07:51:25 localhost kubelet-wrapper[1310]: W0410 07:51:25.681627    1310 server.go:445] Could not load kubeconfig file /var/lib/kubelet/kubeconfig: stat /var/lib/kubelet/kubeconfig: no such file or directory. Trying auth path instead.
Apr 10 07:51:25 localhost kubelet-wrapper[1310]: W0410 07:51:25.682129    1310 server.go:406] Could not load kubernetes auth path /var/lib/kubelet/kubernetes_auth: stat /var/lib/kubelet/kubernetes_auth: no such file or directory. Continuing with defaults.
Apr 10 07:51:25 localhost kubelet-wrapper[1310]: I0410 07:51:25.682202    1310 plugins.go:71] No cloud provider specified.
Apr 10 07:51:25 localhost kubelet-wrapper[1310]: I0410 07:51:25.682639    1310 manager.go:132] cAdvisor running in container: "/system.slice/kubelet.service"
Apr 10 07:52:13 localhost kubelet-wrapper[1310]: I0410 07:52:13.926526    1310 fs.go:109] Filesystem partitions: map[/dev/sda9:{mountpoint:/var/lib/docker major:8 minor:9 fsType:ext4 blockSize:0} /dev/sda3:{mountpoint:/etc/ssl/certs major:8 minor:3 fsType: blockSize:0} ov
Apr 10 07:52:16 localhost kubelet-wrapper[1310]: I0410 07:52:16.684915    1310 manager.go:169] Machine: {NumCores:1 CpuFrequency:2294581 MemoryCapacity:1045590016 MachineID:c0bb99276bc74b86b88d83396764e3f7 SystemUUID:6E4687DC-F9AD-47EA-8CA3-2B4FD8DB4F86 BootID:44a5ed53-4e
Apr 10 07:52:16 localhost kubelet-wrapper[1310]: I0410 07:52:16.688227    1310 manager.go:175] Version: {KernelVersion:4.5.0-coreos-r1 ContainerOsVersion:Debian GNU/Linux 8 (jessie) DockerVersion:1.10.3 CadvisorVersion: CadvisorRevision:}
Apr 10 07:52:16 localhost kubelet-wrapper[1310]: I0410 07:52:16.692075    1310 server.go:673] Adding manifest file: /etc/kubernetes/manifests
Apr 10 07:52:16 localhost kubelet-wrapper[1310]: I0410 07:52:16.692106    1310 server.go:683] Watching apiserver
```
Do I miss some steps? I have redone this for several times, but always get this error.

**Comment by colhom** [comment]
@Leorowe if you `docker ps`, do you see the apiserver container running? Can you curl `localhost:8080/api` successfully while ssh'd into the vagrant vm?

**Comment by Leorowe** [comment]
Hi,
No, there is no any container was running and connection refused when curl localhost:8080/api

**Comment by chrisgeo** [comment]
Same thing happens on the multi-node setup as well.

**Comment by bfallik** [comment]
We recently encountered this as well. The [instructions](https://coreos.com/kubernetes/docs/latest/kubernetes-on-vagrant-single.html#start-the-machine) recommend exporting KUBECONFIG:
```
$ export KUBECONFIG="${KUBECONFIG}:$(pwd)/kubeconfig"
```
which I think is a recent change that seems to not actually work with kubectl. Our workaround is to unset KUBECONFIG. I don't know why it's necessary since (I think) the set-context calls record the config in ~/.kube/config for use by kubectl.

**Comment by jamesgoodhouse** [comment]
I'm able to get things working if I drop the vm box_version from '1029.0.0' to '1010.1.0'.

**Comment by Leorowe** [comment]
Hi,

Finally, I figured out this issue is caused by the network problem. Because of GFW in China, these images couldn't be downloaded. After setting proxy, everything work fine.

> image: downloading signature from https://quay.io/c1/aci/quay.io/coreos/hyperkube/v1.2.0_coreos.1/aci.asc/linux/amd64/
Apr 30 11:10:44 localhost kubelet-wrapper[1205]: Downloading signature:  0 B/473 B
Apr 30 11:10:44 localhost kubelet-wrapper[1205]: Downloading signature:  473 B/473 B
Apr 30 11:11:15 localhost kubelet-wrapper[1205]: run: Get https://quay-registry.s3.amazonaws.com/sharedimages/bbf48108-4be9-43d0-8b3a-d02cdf16a459/layer?Signature=xYf%2BFBovmAG3PFWUqwP6XuPId8o%3D&Expires=1462014736&AWSAccessKeyId=AKIA
Apr 30 11:11:15 localhost systemd[1]: kubelet.service: Main process exited, code=exited, status=1/FAILURE
mage: using image from file /usr/share/rkt/stage1-fly.aci
Apr 30 11:11:25 localhost kubelet-wrapper[1311]: image: searching for app image quay.io/coreos/hyperkube
Apr 30 11:11:27 localhost kubelet-wrapper[1311]: image: remote fetching from URL "https://quay.io/c1/aci/quay.io/coreos/hyperkube/v1.2.0_coreos.1/aci/linux/amd64/"
Apr 30 11:11:27 localhost kubelet-wrapper[1311]: image: keys already exist for prefix "quay.io/coreos/hyperkube", not fetching again
Apr 30 11:11:27 localhost kubelet-wrapper[1311]: image: downloading signature from https://quay.io/c1/aci/quay.io/coreos/hyperkube/v1.2.0_coreos.1/aci.asc/linux/amd64/
Apr 30 11:11:28 localhost kubelet-wrapper[1311]: Downloading signature:  0 B/473 B
Apr 30 11:11:28 localhost kubelet-wrapper[1311]: Downloading signature:  473 B/473 B
Apr 30 11:11:59 localhost kubelet-wrapper[1311]: run: Get https://quay-registry.s3.amazonaws.com/sharedimages/bbf48108-4be9-43d0-8b3a-d02cdf16a459/layer?Signature=qjsqRILHR%2B9GM6kg2SPdlQwmWEU%3D&Expires=1462014779&AWSAccessKeyId=AKIA
Apr 30 11:11:59 localhost systemd[1]: kubelet.service: Main process exited, code=exited, status=1/FAILURE
Apr 30 11:11:59 localhost systemd[1]: kubelet.service: Unit entered failed state.
Apr 30 11:11:59 localhost systemd[1]: kubelet.service: Failed with result 'exit-code'.

**Comment by jamesgoodhouse** [comment]
@Leorowe, what, if any, changes did you have to make? I am still unable to make this work and have had to revert the Vagrantfile to use `1010.1.0`.

**Comment by Leorowe** [comment]
Actually I just set the HTTP proxy. May be you should use `journalctl -u kubelet` to check whether these is any exception log?

**Comment by drbgd** [comment]
I've got the same problem - tried destroying the vagrant setup and re-provisioning around 3 times on a bad/flaky connection, and ran into this every time. After the 230MB image was downloaded, I also got an error about "Missing required config value: ETCD_ENDPOINTS" on c1.

I reprovisioned using a good/stable connection and now it works. Not sure if this is a problem of having to wait longer on a poor connection or not - but on the good connection, I didn't get any error related to ETCD_ENDPOINTS.
